packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -59,34 +60,34 @@ export class ProxyTarget {
     return new ProxyTarget(undefined, cluster);
   }
 
-  private constructor(private readonly dbInstance?: IDatabaseInstance, private readonly dbCluster?: IDatabaseCluster) {}
+  private constructor(
+    private readonly dbInstance: IDatabaseInstance | undefined,
+    private readonly dbCluster: IDatabaseCluster | undefined) {
+  }
```

Comment:
The problem with this solution is that below, we need to actually differentiate between the two types when we set either `dbClusters` or `dbInstances`. And since `IDatabaseInstance` and `IDatabaseCluster` are interfaces, we can't use `instanceof` for them.

So I'd rather leave this as-is.

Reply:
Ah, I see now. `ProxyTargetConfig` needs to differentiate so we can pass the right values to the `CfnDBProxyTargetGroup`. Got it. :+1: